src/Mod/Assembly/UtilsAssembly.py

```diff
@@ -212,6 +196,29 @@ def getObject(ref):
     return None
 
 
+def process_body(body, names, i):
+    # If there's a next name, it could either be elementName, in which case we can return the Body.
+    # But it could be a subobject like Sketch or datums.
+    if i + 1 < len(names):
+        obj2 = None
+        for obji in body.OutList:
+            if obji.Name == names[i + 1]:
+                obj2 = obji
+                break
```

Comment:
I would prefer if the `name` assignment was moved out of the for loop, right next to the length check of `names`. This to make it easier to read.
```suggestion
    if i + 1 < len(names):
        name = names[i + 1]
        obj2 = None
        for obji in body.OutList:
            if obji.Name == name:
                break
```